tests/components/normal-components/chip-pin-alias.test.tsx

```diff
@@ -0,0 +1,29 @@
+import { expect, it } from "bun:test"
+import "lib/register-catalogue"
+import { getTestFixture } from "tests/fixtures/get-test-fixture"
+
+it("Chip with pins in string", async () => {
+  const { circuit } = getTestFixture()
+
+  circuit.add(
+    <board width="10mm" height="10mm">
+      <chip
+        name="U1"
+        schPinSpacing={0.75}
+        schPortArrangement={{
+        leftSide: {
+          direction: "top-to-bottom",
+          pins: ["B1", "B2"],
```

Comment:
I think there's a bug in your implementation, it looks like you assume `pin1` is `B1` because of the integer in the string- we shouldn't do this. In fact, usually in circuits with `B1` there is also an `A1`. Can you modify this test 
```suggestion
          pins: ["B1", "B2"],
        },
        rightSide: { direction: "top-to-bottom", pins: ["A1", "A2"] }
```

Then check the output in the test to ensure that there are four schematic_port.pin_number


My hunch is that a lot of this PR will need to be redone because I think you might have changed code that specifically handled aliases, but I'm not completely sure

Reply:
@seveibar Actually, I am not assuming the `pin1` is `B1` because of the integer in the string. I am just incrementing the pinNumbers as new ports are getting added. The `B1` here is just an alias for that pin, so now I have updated the pin label as the alias, or default to pin number (will have to update the circuit_json with the variable name). Take a look at the updated test file and snap.

Reply:
I don't understand- `pin_number` should always be a number. `B1`/`A1` is a pin label

Reply:
Yeah, that's what I will update in circuit_json as well. Just wanted to confirm that the logic is not taking the string's Integer as pin number